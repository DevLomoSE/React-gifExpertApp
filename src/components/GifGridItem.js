import React from 'react';
import PropTypes from "prop-types";

const GifGridItem = ({id, title, img}) => {

  return (
    <div className="card animate__animated animate__heartBeat animate__zoomInUp">
      <img src={img} alt={title}></img>
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired
};

export default GifGridItem;
