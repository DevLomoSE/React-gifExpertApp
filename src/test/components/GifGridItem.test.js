import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";
import "@testing-library/jest-dom";

import "../../setupTests";

describe("testing for <GifGridItem />", () => {

  const titulo = 'titulo';
  const url = "https://4.bp.blogspot.com/-huuOLwohWYo/UngFPvvlpHI/AAAAAAAAb_o/DObAVnXFESo/s1600/hobbit-desolation-of-smaug-tauriel-poster.jpg";
  const wrapper = shallow(<GifGridItem 
    title={titulo}
    img={url} />);

  test("should render gifGridItem", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render 'titulo'", () => {
    let p = wrapper.find('p').text().trim();

    expect(p).toBe(titulo);
  });

  test("img should have url and alt", () => {
    let img = wrapper.find('img');
    
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(titulo);
  });

  test('should have animated__fadeIn', () => {
    let div = wrapper.find('div');
    
    expect(div.prop("className")).toMatch(/animate__zoomInUp/);
    expect(div.hasClass("animate__zoomInUp")).toBe(true);
  });

});
