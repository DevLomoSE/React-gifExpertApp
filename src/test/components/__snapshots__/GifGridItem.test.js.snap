// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for <GifGridItem /> should render gifGridItem 1`] = `
<div
  className="card animate__animated animate__heartBeat animate__zoomInUp"
>
  <img
    alt="titulo"
    src="https://4.bp.blogspot.com/-huuOLwohWYo/UngFPvvlpHI/AAAAAAAAb_o/DObAVnXFESo/s1600/hobbit-desolation-of-smaug-tauriel-poster.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
